numba/_dispatcher.cpp

```diff
@@ -530,14 +586,10 @@ Dispatcher_call(DispatcherObject *self, PyObject *args, PyObject *kws)
         }
     }
 
-    /* If compilation is enabled, ensure that an exact match is found and if
-     * not compile one */
-    int exact_match_required = self->can_compile ? 1 : self->exact_match_required;
-
     /* We only allow unsafe conversions if compilation of new specializations
        has been disabled. */
-    cfunc = dispatcher_resolve(self->dispatcher, tys, &matches,
-                               !self->can_compile, exact_match_required);
+    cfunc = self->resolve(tys, matches, !self->can_compile,
```

Comment:
Perhaps leave a comment about `matches` being mutated? It's not obvious from the call site any more.

Reply:
Yes, good point.

Reply:
Added.

Reply:
thanks